### PR TITLE
contrib: app init improvements

### PR DESCRIPTION
**Pull request by 0xcryptovenom**
- consolidate the app's configuration resolution / initialization in its own `initConfig` function.
- move the config's resolution / initialization + persistence from an empty-dependencies `useEffect()` + `setState()` (client-side only) outside of the App component.
  - the `initConfig` must now return `null` on the server to avoid type errors (resolution of the config relies on the global `window`), and to avoid risk of having a single configuration in the server-side process for all users - if the config is resolved & initialized on the server, and it should, then it must be user/request-specific.
  - instead of having to wait for the first render & mount of of the App component before dealing with the configuration, it's now initialized immediately when the JS bundle is evaluated, and the config is available for the first render.
  - as a result the rendering & side-effects sequence of the application's initialization is more streamlined & efficient even though unlikely to provide a perceptible performance user experience improvement yet. 
- replace the double ~redundant app initialization local component states with a single state-machine-ish `initStatus` which better reflects the different stages of the app's initialisation.
- finally, remove the now-dead `Pause` component & associated code path for improved clarity over the code base.
  
The same pattern could be applied in many places in the code base, especially for static / constant objects, configurations which should can easily be moved outside of the render path.
  
More comments have been added to reflect the ideal direction that is not necessarily immediately available.

- improve concurrency of `AdrenaClient.initialize` async operations: loading `cortex` does not need to block loading `custodies`.

### CODE REVIEW

**Comment by 0xcryptovenom** [comment]
Updated to provide a reasonably detailed PR description.

**Comment by adrena-orex** [comment]
Thank you for the PR, looking into it

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -49,14 +46,21 @@ function Loader(): JSX.Element {
...
-  const [config, setConfig] = useState<IConfiguration | null>(null);
-  const [isInitialized, setIsInitialized] = useState<boolean>(false);
-  const [initializationInProgress, setInitializationInProgress] =
-    useState<boolean>(false);
-
+  const [initStatus, setInitStatus] = useState<
+    'not-started' | 'starting' | 'done'
+  >('not-started');
+  // FIXME: RPC selection shouldn't be hook / effect-based.
+  // A default RPC should be picked by the server durring ssr.
```
[the remark this answers is not in the document]
Comment (0xcryptovenom) [reply src/pages/_app.tsx:63]: Need to find how to leverage Next.js' mechanism to run the config + RPC logic during SSR, scoped to the current HTTP request where we can read preferences stored in cookies.

Then pass it as some serialized initial state into a some sort of a global variable that is picked up by the client when it boots.

A good first step is to keep that in mind, but only improve the client-side-only implementation, not using hooks / useEffect, similar to what was done for the config resolution.

[the remark this answers is not in the document]
Comment (0xcryptovenom) [reply src/pages/_app.tsx:63]: https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props

[the remark this answers is not in the document]
Comment (0xcryptovenom) [reply src/pages/_app.tsx:71]: Not sure I understood your comment correctly the first time, but so far we don't really leverage SSR, as the initialization / rendering of the app is conditioned to an effect updating local state.

**Comment by 0xcryptovenom** [comment]
Rebased, but needs to be double-checked / tested

**Pull request merged**